source/rafcon/gui/shortcut_manager.py

```diff
@@ -135,20 +136,21 @@ def get_shortcut_for_action(self, action):
             return self.__action_to_shortcuts[action]
         return None
 
-    def trigger_action(self, action, key_value, modifier_mask):
+    def trigger_action(self, action, key_value, modifier_mask, cursor_position=None):
```

Comment:
Okay, i implement this, but it will take a while, because i have to refactor all callbacks registered in the shortcut_manager.

Reply:
I think this is not necessary. Please check before changing every callback!

Reply:
Currently every callback that, wasn't refactored throws an error. 
Example: 

`ERROR - rafcon.gui.shortcut_manager:  Exception while calling callback methods for action "delete": _delete_selection() got an unexpected keyword argument 'cursor_position'`

Reply:
Ah I see the reason. In line 60 of `shortcut_manager.py`, you always pass the `cursor_position` to `self.trigger_action`. I am not sure, whether we should do this. as you say, this would require _all_ registered callbacks to be changed. The information about the cursor is _globally_ available (`GtkWindow.get_pointer()`), so there is no need to pass it as parameter. Every callback that requires this info can get it itself easily.

`trigger_action` can also be called from outside the shortcut manager. In fact, this is often done by a menu entry callback.

Nevertheless, adding `**kwargs` to all callbacks would allow us to pass arbitrary additional parameters to callbacks, which could be of advantague, as this example shows.

Maybe @sebastian-brunner can share his opinion on that, too!

Reply:
Yes, but the current cursor_position was not the reason, why i extended the event tuple. 
When using the right-click menu, content has to be pasted at the position of the right-click menu, and not the current cursor_position. So i had to pass it, and therefore extended the event.

Reply:
Good point, one more reason to add `**kwargs` to all callbacks. Still, I would like to hear @sebastian-brunner's opinion.

Reply:
Having `**kwargs` in the trigger_action method is definitely a nice way to provide extensibility in the future. So I would go for that as well.

Reply:
Okay, so i will proceed, and add `**kwargs` to all callbacks.